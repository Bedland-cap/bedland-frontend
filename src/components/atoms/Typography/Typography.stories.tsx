import type { Meta, StoryObj } from '@storybook/react';
import { colorNames, typographyNames } from 'theme/theme.types';
import ThemeProvider from 'theme/ThemeContext';
import Typography from './Typography';

const meta = {
  title: 'Basic Components/Atoms/Typography',
  component: Typography,
  tags: ['autodocs'],
  argTypes: {
    variant: {
      options: typographyNames,
      control: 'radio',
      description: 'Typography variant',
      defaultValue: { summary: 'paragraph' },
      table: {
        type: {
          summary: `Variants: ${typographyNames.toString().replaceAll(',', ' | ')}`,
        },
        defaultValue: { summary: 'paragraph' },
      },
    },
    children: {
      name: 'label',
      description: 'Text to display',
    },
    style: {
      table: {
        disable: true,
      },
    },
    color: {
      options: colorNames,
      control: 'select',
      description: 'Color of the text',
      defaultValue: 'default',
    },
  },
  decorators: [(Story) => <ThemeProvider>{Story()}</ThemeProvider>],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof Typography>;

export default meta;
type Story = StoryObj<typeof meta>;

export const PageTitle: Story = {
  args: {
    variant: 'pageTitle',
    color: 'default',
    children: 'Page Title',
  },
};

export const Paragraph: Story = {
  args: {
    variant: 'paragraph',
    color: 'default',
    children:
      'There was nothing so very remarkable in that; nor did Alice think it so very much out of the way to hear the Rabbit say to itself, - Oh dear! Oh dear! I shall be late!- (when she thought it over afterwards, it occurred to her that she ought to have wondered at this, but at the time it all seemed quite natural); but when the Rabbit actually took a watch out of its waistcoat-pocket, and looked at it, and then hurried on, Alice started to her feet, for it flashed across her mind that she had never before seen a rabbit with either a waistcoat-pocket, or a watch to take out of it, and burning with curiosity, she ran across the field after it, and fortunately was just in time to see it pop down a large rabbit-hole under the hedge. ',
  },
};

export const Caption: Story = {
  args: {
    variant: 'caption',
    color: 'default',
    children: 'Caption',
  },
};

export const SidebarItem: Story = {
  args: {
    variant: 'sidebarItem',
    color: 'default',
    children: 'Sidebar Item',
  },
};
