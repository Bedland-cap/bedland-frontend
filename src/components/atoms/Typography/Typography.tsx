import { CSSProperties, PropsWithChildren, useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import { ColorNames, TypographyName } from 'theme/theme.types';

const variantToElement = {
  pageTitle: 'h2',
  sectionTitle: 'h2',
  sidebarItem: 'span',
  paragraph: 'p',
  caption: 'p',
  widgetTitle: 'h5',
  cardTitle: 'h3',
  messageItem: 'h4',
  error: 'p',
  subHeader: 'h6',
  header5: 'h5',
  logout: 'p',
  notificationItem:'h4'
} as const;

export type TypographyProps = {
  variant?: TypographyName;
  color?: ColorNames;
  style?: CSSProperties;
};

const defaultProps: TypographyProps = {
  variant: 'paragraph',
  color: 'default',
  style: {},
};

const Typography = ({ variant, style, color, children }: PropsWithChildren<TypographyProps>) => {
  const { palette, typography } = useContext(ThemeContext);
  const Element = variantToElement[variant ?? 'paragraph'];

  const colorStyles: CSSProperties = {
    color: color ? palette[color] : 'none',
  };

  return (
    <Element
      style={{
        ...typography[variant ?? 'paragraph'],
        ...colorStyles,
        ...style,
      }}
    >
      {children}
    </Element>
  );
};

Typography.defaultProps = defaultProps;

export default Typography;
