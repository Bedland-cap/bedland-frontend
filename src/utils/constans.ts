export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
