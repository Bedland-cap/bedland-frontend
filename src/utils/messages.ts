const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
