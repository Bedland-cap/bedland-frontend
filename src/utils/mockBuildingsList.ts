import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
