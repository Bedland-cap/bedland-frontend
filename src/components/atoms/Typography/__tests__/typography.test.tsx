import { PropsWithChildren } from 'react';
import { render } from 'test.utils';
import Typography, { TypographyProps } from '../Typography';

describe('Typography', () => {
  const item = ({ variant, color, children }: PropsWithChildren<TypographyProps>) =>
    render(
      <Typography variant={variant} color={color}>
        {children}
      </Typography>,
      {},
    );

  it('should renders correctly typography - paragraph', () => {
    expect(
      item({ variant: 'paragraph', color: 'primaryDark', children: 'Paragraph' }),
    ).toMatchSnapshot();
  });

  it('should renders correctly typography - sidebarItem', () => {
    expect(
      item({ variant: 'sidebarItem', color: 'primaryDark', children: 'SidebarItem' }),
    ).toMatchSnapshot();
  });

  it('should renders correctly typography - caption', () => {
    expect(
      item({ variant: 'caption', color: 'primaryDark', children: 'Caption' }),
    ).toMatchSnapshot();
  });
});
