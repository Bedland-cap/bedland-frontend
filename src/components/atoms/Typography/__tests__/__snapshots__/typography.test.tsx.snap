// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should renders correctly typography - caption 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
      >
        Caption
      </p>
      <div
        class="sc-bcXHqe hRlqBY"
      >
        <div
          class="sc-gswNZR kKeZN"
        >
          <h2
            style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
          >
            Welcome Trish!
          </h2>
        </div>
        <div
          class="sc-dkrFOg dLa-DCJ"
        />
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
    >
      Caption
    </p>
    <div
      class="sc-bcXHqe hRlqBY"
    >
      <div
        class="sc-gswNZR kKeZN"
      >
        <h2
          style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
        >
          Welcome Trish!
        </h2>
      </div>
      <div
        class="sc-dkrFOg dLa-DCJ"
      />
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Typography should renders correctly typography - paragraph 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
      >
        Paragraph
      </p>
      <div
        class="sc-bcXHqe hRlqBY"
      >
        <div
          class="sc-gswNZR kKeZN"
        >
          <h2
            style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
          >
            Welcome Trish!
          </h2>
        </div>
        <div
          class="sc-dkrFOg dLa-DCJ"
        />
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
    >
      Paragraph
    </p>
    <div
      class="sc-bcXHqe hRlqBY"
    >
      <div
        class="sc-gswNZR kKeZN"
      >
        <h2
          style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
        >
          Welcome Trish!
        </h2>
      </div>
      <div
        class="sc-dkrFOg dLa-DCJ"
      />
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Typography should renders correctly typography - sidebarItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; text-transform: capitalize; color: rgb(69, 83, 221);"
      >
        SidebarItem
      </span>
      <div
        class="sc-bcXHqe hRlqBY"
      >
        <div
          class="sc-gswNZR kKeZN"
        >
          <h2
            style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
          >
            Welcome Trish!
          </h2>
        </div>
        <div
          class="sc-dkrFOg dLa-DCJ"
        />
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; text-transform: capitalize; color: rgb(69, 83, 221);"
    >
      SidebarItem
    </span>
    <div
      class="sc-bcXHqe hRlqBY"
    >
      <div
        class="sc-gswNZR kKeZN"
      >
        <h2
          style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(0, 0, 0);"
        >
          Welcome Trish!
        </h2>
      </div>
      <div
        class="sc-dkrFOg dLa-DCJ"
      />
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
